example.config.yml

```diff
@@ -1,64 +1,78 @@
 version: 1
 cloud:
-  provider: aws-ecs
+  provider: aws-ecs # Are you thinking there might be multiple provider types on top of AWS? Like aws-ec2 or something?
   ecs_cluster_name: singer
   bucket_name: some-bucket
+  docker_base_image: image:tag # optional; will use this base image or default to python:3.8 if not provided
+  dockerfile: <path> # optional; will prepend this dockerfile onto the DOCKERFILE_TEMPLATE in docker.py
+  configs-path: configs # optional; default value is configs. Will put configs into relative dir configs. The run-pipeline command will look here. The push and pull commands will look here to sync
+  catalogs-path: catalogs # optional
+  states-path: states # optional
 taps:
   tap-salesforce:
-    repo: git@github.com:singer-io/tap-salesforce.git
-    default_singer_config:
+    package: tap-salesforce==1.4.32 # Specify any valid pip-installable package reference or repository. This format would look in pypi. Version is optional and used directly by pip. E.g. "pip install tap-salesforce==1.4.32"
```

Comment:
Since many taps are not even in pypi, starting off with `repo` support, eventually supporting `package` as well. Repo would support all taps / targets, `package` would only support those in pypi.